subsys/bluetooth/host/conn.c

```diff
@@ -1643,6 +1646,12 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 			/* Cancel Connection Update if it is pending */
 			if (conn->type == BT_CONN_TYPE_LE) {
 				k_delayed_work_cancel(&conn->update_work);
+#if defined(CONFIG_BT_REMOTE_VERSION)
+				if (atomic_test_and_clear_bit(conn->flags,
```

Comment:
Actually, why are you doing this in a `if( conn->type == BT_CONN_TYPE_LE)` branch? I don't think we limited the version request to LE-only connections. You'll also end up with a shorter line if you move it out from that branch.

Update: I now see that I overlooked this as well - you seem to be doing it only for LE even though it's not an LE-specific procedure. See my other comments.